### PR TITLE
Support Android 11's new file access permission

### LINKED ISSUE #898: [BUG] Can't enter a writable file path above v107 / why no graphical folder seclection?
Galaxy S10 (And 11)

Since v107, I can't enter a writable file path. I don't want to use the hidden default folder but I can't select a simple folder like "Downloads". Folders on an SD card don't work either, all permissions are granted.
Why do we still have to enter the path manually? For people (like me), who find paths on Android more than confusing, it's a real pain. Why not the standard graphical selection as in every other app?

### PR DESCRIPTION
**Pull request by blastrock**
Fixes #898

### CODE REVIEW

File: gpslogger/src/main/java/com/mendhak/gpslogger/ui/fragments/settings/LoggingSettingsFragment.java [hunk 1]
```diff
@@ -166,12 +169,12 @@ public boolean onPreferenceChange(final Preference preference, Object newValue)
             }
             catch(Exception e){
                 LOG.error("Could not create chosen directory path", e);
-                Dialogs.error(getString(R.string.error), getString(R.string.pref_logging_file_no_permissions), e.getMessage(), e, getActivity());
+                displayFileAccessError();
```
Comment (blastrock) [inline gpslogger/src/main/java/com/mendhak/gpslogger/ui/fragments/settings/LoggingSettingsFragment.java:172]: Note: during my tests, only the error below with `isAllowedToWriteTo` showed. I don't know if we really need to change this.

**Comment by mendhak** [comment]
Fixes #880 
Fixes #886

**Comment by mendhak** [comment]
@blastrock thank you!!  I didn't even realize this was an option, so simple.    I had gone down the path of scoped storage in the Android 11 docs and got stuck on how I'd manage multiple file extensions at the same time (because you can write to .csv and .gpx at the same time).  But what you've done here makes so much sense.  I'm going to test this out.

File: gpslogger/src/main/java/com/mendhak/gpslogger/ui/fragments/settings/LoggingSettingsFragment.java (hunk 1 above)
Comment (mendhak) [reply gpslogger/src/main/java/com/mendhak/gpslogger/ui/fragments/settings/LoggingSettingsFragment.java:181]: I did some testing and ran into both situations.  One with the directory not being created.  The other was `isAllowedToWriteTo` was returning _true_ but when the app tried to create a file later, it failed.  So I've added a very deliberate test here to create a file then delete it.

**Comment by blastrock** [comment]
Well, you're welcome :)
I know how Android APIs can be so confusing and hard to use...

I just reviewed you two additional patches, they look fine to me. Thanks for the testing and all the good work you put into this app :)

**Pull request merged**